### PR TITLE
Version Updates: Node v16, React v17, & some others...

### PR DESCRIPTION
**Pull request by alexkb0009**
Once merged, everyone will be asked to upgrade their local environment to Node 16.

----

(**Tangential**:) Will probably try to work in webpack 5 changes in a subsequent branch/pr, since those seem to [add some NPM packages](https://github.com/4dn-dcic/fourfront/pull/1593/files) that could/should be audited.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -115,21 +115,27 @@
...
     "react-tooltip": "4.2.10",
-    "react-transition-group": "^4.4.1",
+    "react-transition-group": "^4.4.2",
     "redux": "^3.7.0",
-    "subprocess-middleware": "0.1.0",
+    "subprocess-middleware": "0.2.0",
     "underscore": "^1.10.2",
     "url": "^0.11.0"
   },
+  "overrides": {
```
Comment (alexkb0009) [inline package.json:131]: Doesn't seem to be working, at least in this state. Might try to revert to explicit version numbers, but ultimately probably revert `overrides`.

File: package.json [hunk 2]
```diff
@@ -87,9 +87,9 @@
   },
   "dependencies": {
     "@fortawesome/fontawesome-free": "^5.14.0",
-    "@hms-dbmi-bgm/react-workflow-viz": "0.1.4",
-    "@hms-dbmi-bgm/shared-portal-components": "github:4dn-dcic/shared-portal-components#0.1.34",
-    "auth0-lock": "^11.30.4",
+    "@hms-dbmi-bgm/react-workflow-viz": "0.1.5",
+    "@hms-dbmi-bgm/shared-portal-components": "github:4dn-dcic/shared-portal-components#0.1.36b2",
```
Comment (alexkb0009) [inline package.json:91]: TODO/self-reminder: Get SPC reviewed (will link it to this PR soon), and update this before ready to merge

File: package.json (hunk 1 above)
Comment (alexkb0009) [reply package.json:131]: 'npm link' seems to work better, without having to do the symlink script, but still double-checking. If consistently works then will remove that part from the docs.

File: package.json (hunk 2 above)
Comment (alexkb0009) [reply package.json:91]: 'npm link' seems to work better w. latest Node+NPM versions, without having to do the symlink script, but still double-checking. If consistently works then will remove that part from the docs.

File: src/encoded/static/components/viz/PedigreeViz/IndividualNodeShapeLayer.js [hunk 3]
```diff
@@ -194,7 +194,7 @@ const AffectedBGPieChart = React.memo(function AffectedBGPieChart({
         return diseaseToIndex[disease];
     });
     const diseaseLen = visibleDiseases.length;
-    if (diseaseLen === 0) return;
+    if (diseaseLen === 0) return null;
```
Comment (alexkb0009) [inline src/encoded/static/components/viz/PedigreeViz/IndividualNodeShapeLayer.js:197]: Probably the only breaking edit so far from React 17 update. Need to return `null` from component that wants to render nothing, `undefined` is no good.

Comment (alexkb0009) [reply src/encoded/static/components/viz/PedigreeViz/IndividualNodeShapeLayer.js:197]: TODO/self-reminder: Test react-workflow-viz view(s).

File: package.json (hunk 1 above)
Comment (alexkb0009) [reply package.json:131]: Same result, reverting,..

**Review by willronchetti — APPROVED** [review]
Feel free to merge once tested on devtest

**Review by Bianca-Morris — APPROVED** [review]
I haven't been able to find any other issues with this, so provided things are also working well for Utku, I'd say we should give it a go. Also, thanks for cleaning up some of those console warnings, it's looking better already.

**Review by utku-ozturk — APPROVED** [review]
Looks great to me! (not tested it locally but we ported most of the changes into FF and it works good there.)

**Pull request merged**